src/components/button/button.spec.js

```diff
@@ -0,0 +1,364 @@
+import React from 'react';
+import 'jest-styled-components';
+import { mount, shallow } from 'enzyme';
+import Icon from 'components/icon';
+import Link from 'components/link';
+import TestRenderer from 'react-test-renderer';
+import Button from '.';
+import StyledButton from './button.style';
+import BaseTheme from '../../style/themes/base';
+import classicTheme from '../../style/themes/classic';
+
+import { assertStyleMatch } from '../../__spec_helper__/test-utils';
+import { rootTagTest } from '../../utils/helpers/tags/tags-specs';
+
+const render = (props, renderer = shallow) => {
+  return renderer(
+    <Button { ...props } />
+  );
+};
+
+const variants = ['primary', 'secondary', 'tertiary', 'destructive', 'darkBackground'];
+const sizes = { small: [32, 16], medium: [40, 24], large: [48, 32] };
+
+describe('Button', () => {
+  describe('when no props other than children are passed into the component', () => {
+    it('renders the default props and children to match the snapshot', () => {
+      const wrapper = render({ children: 'foo' });
+      expect(wrapper.props().renderAs).toEqual('secondary');
+      expect(wrapper.props().size).toEqual('medium');
+      expect(wrapper.props().disabled).toEqual(false);
+      expect(wrapper.props().iconPosition).toEqual('');
+      expect(!wrapper.contains(<Icon type='filter' />)).toBeTruthy();
```

Comment:
why is that exclamation mark for?

Reply:
`toBeEdgy` 😆
Not sure what I was doing there, I've removed it and changed the expectation to `toBeFalsy()`